### ISSUE TITLE
Azure Deployment Fails

### ISSUE THREAD

**Issue by DerekTBrown** [docker/hub-feedback#1296]
A problem occurs when I try to deploy to Azure:

```
ERROR: 400 Client Error: Bad Request for url: https://management.azure.com/subscriptions/fbea5a02-c254-4899-9660-1dbe84356159/resourcegroups/project1/providers/Microsoft.Resources/deployments/project1?api-version=2016-09-01: {u'error': {u'message': u'Marketplace purchase eligibilty check returned errors. See inner errors for details. ', u'code': u'MarketplacePurchaseEligibilityFailed', u'details': [{u'message': u'Offer with PublisherId: docker, OfferId: docker-ce cannot be purchased due to validation errors. See details for more information.[{"Legal terms have not been accepted for this item on this subscription. To accept legal terms using PowerShell, please use Get-AzureRmMarketplaceTerms and Set-AzureRmMarketplaceTerms API(https://go.microsoft.com/fwlink/?linkid=862451) or deploy via the Azure portal to accept the terms":"StoreApi"}]', u'code': u'BadRequest'}, {u'message': u'Offer with PublisherId: docker, OfferId: docker-ce cannot be purchased due to validation errors. See details for more information.[{"Legal terms have not been accepted for this item on this subscription. To accept legal terms using PowerShell, please use Get-AzureRmMarketplaceTerms and Set-AzureRmMarketplaceTerms API(https://go.microsoft.com/fwlink/?linkid=862451) or deploy via the Azure portal to accept the terms":"StoreApi"}]', u'code': u'BadRequest'}]}}

```

**Comment by nh2d** [comment]
+1 still occuring

**Comment by lpassig** [comment]
Have a look at: https://azure.microsoft.com/en-us/blog/working-with-marketplace-images-on-azure-resource-manager/ You need to accept/allow the programatic deployment fpr Docker either via the portal or via Powershell. The Cmdlets are only available starting version PS AzureRM Version 5.4.0.
